bootstrap/utils/utils_test.go

```diff
@@ -278,3 +279,37 @@ func TestDeepCopy(t *testing.T) {
 	assert.NotEqual(t, orig, clone)
 
 }
+
+func TestLoadFile(t *testing.T) {
+	// Types
+	var configuration map[string]any
+
+	tests := []struct {
+		Name        string
+		Path        string
+		Contents    any
+		ExpectedErr string
+	}{
+		{"Valid - load from YAML file", path.Join("..", "config", "testdata", "configuration.yaml"), configuration, ""},
+		{"Valid - load from JSON file", path.Join(".", "testdata", "configuration.json"), configuration, ""},
+		{"Valid - load from HTTPS", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/testdata/configuration.yaml", configuration, ""},
+		{"Invalid - File not found", "bogus", nil, "Could not read file"},
+		{"Invalid - load invalid file type", path.Join("..", "bootstrap.go"), nil, "Could not load unknown file type"},
+		// {"Invalid - load from YAML file", path.Join("..", "config", "testdata", "bogus.yaml"), configuration, "Could not unmarshal YAML"},
+		{"Invalid - load from JSON file", path.Join(".", "testdata", "bogus.json"), configuration, "Could not unmarshal JSON"},
+		{"Invalid - load from invalid HTTPS", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/configuration.yaml", configuration, "Invalid status code"},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			err := LoadFile(tc.Path, &tc.Contents)
+			if tc.ExpectedErr != "" {
+				assert.Contains(t, err.Error(), tc.ExpectedErr)
+				return
+			}
+
+			assert.NotNil(t, tc.Contents)
+			tc.Contents = make(map[string]any)
```

Comment:
Not sure what this is doing?

Reply:
this was resetting the configuration information - so that the variable can be reused - but this will likely get refactored out if we want to keep the signature of the LoadFile the same as ReadFile - just thought it would be nice to implement the marshaling here so that we can get rid of a lot of similar code.